### PR TITLE
Kill actor child processes on shutdown

### PR DESCRIPTION
**Pull request by ericl**
It's common to spawn child processes from actor with the expectation that they fate share with the actor. However currently terminating the actor usually doesn't kill the children.

This patch changes this by setting the process group id of actors, and sending SIGTERM to the entire process group on actor shutdown. Since the worker has an exit hook, this cleanly quits the actor and also runs the atexit hooks.

Related: https://github.com/ray-project/ray/issues/3295

### CODE REVIEW

File: python/ray/worker.py [hunk 1]
```diff
@@ -945,12 +949,14 @@ def _wait_for_and_process_task(self, task):
...
                 title = "ray_worker:{}()".format(function_name)
+                next_title = "ray_worker"
             else:
                 actor = self.actors[task.actor_id().id()]
                 title = "ray_{}:{}()".format(actor.__class__.__name__,
                                              function_name)
+                next_title = "ray_{}".format(actor.__class__.__name__)
             with profiling.profile("task", extra_data=extra_data, worker=self):
-                with _changeproctitle(title):
+                with _changeproctitle(title, next_title):
```
Comment (ericl) [inline python/ray/worker.py]: Actor titles shouldn't revert to `ray_worker` on idle. This is fixed by explicitly setting them to the idle actor name.

**Comment by ericl** [comment]
Ping.

File: python/ray/actor.py [hunk 2]
```diff
@@ -757,8 +759,10 @@ def __ray_terminate__(self):
                 # this is so that when the worker kills itself below, the local
                 # scheduler won't push an error message to the driver.
                 worker.local_scheduler_client.disconnect()
-                import os
-                os._exit(0)
+                # Kill the process group. We will get the SIGTERM and enter
```
Comment (robertnishihara) [inline python/ray/actor.py]:
```suggestion
                # Kill the process group. We will get the SIGTERM and
```

**Review by robertnishihara — APPROVED** [review]
Looks good to me other than the one comment.

**Pull request merged**